CesiumGltfReader/src/GltfReader.cpp

```diff
@@ -422,16 +428,184 @@ Future<GltfReaderResult> GltfReader::resolveExternalData(
           });
 }
 
+bool isKtx(const gsl::span<const std::byte>& data) {
+  const size_t ktxMagicByteLength = 12;
+  if (data.size() < ktxMagicByteLength) {
+    return false;
+  }
+
+  const uint8_t ktxMagic[ktxMagicByteLength] =
+      {0xAB, 0x4B, 0x54, 0x58, 0x20, 0x32, 0x30, 0xBB, 0x0D, 0x0A, 0x1A, 0x0A};
+
+  return memcmp(data.data(), ktxMagic, ktxMagicByteLength) == 0;
+}
+
 /*static*/
-ImageReaderResult
-GltfReader::readImage(const gsl::span<const std::byte>& data) {
-  CESIUM_TRACE("CesiumGltfReader::readImage");
+ImageReaderResult GltfReader::readImage(
+    const gsl::span<const std::byte>& data,
+    const Ktx2TranscodeTargets& ktx2TranscodeTargets) {
+  CESIUM_TRACE("CesiumGltf::readImage");
 
   ImageReaderResult result;
 
   result.image.emplace();
   CesiumGltf::ImageCesium& image = result.image.value();
 
+  if (isKtx(data)) {
+    // TODO: better error handling; make sure KTX-Software doesn't throw
+    // exceptions
```

Comment:
Can it? If so, wrapping in a try/catch is worthwhile so the exception doesn't escape cesium-native.